### PR TITLE
[GPU] Fix permute reorder fusing bfzyx

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Fix permute reorder fusing bfzyx

### Tickets:
 - 123934

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -479,6 +479,10 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, reorder_node
         if (fmt_prev.dimension() > 6 || fmt_next.dimension() > 6)
             return false;
 
+        // Skip reorder fusing to avoid CVS-123934 issue
+        if ((fmt_prev.dimension() == 4) && (fmt_next.dimension() == 5) && (!node.get_users().front()->is_output()))
+            return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:484]: How about preventing this if "allow_new_shape_infer" ?

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:485]: Updated it! Thanks!

**Review by ahnyoung-paul — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 2]
```diff
@@ -479,6 +480,10 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, reorder_node
         if (fmt_prev.dimension() > 6 || fmt_next.dimension() > 6)
             return false;
 
+        // Skip reorder fusing to permute when original permute output has bfxy and fusing output has bfzyx
+        if (allow_new_shape_infer && (fmt_prev.dimension() == 4) && (fmt_next.dimension() == 5) && (!node.get_users().front()->is_output()))
+            return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:485]: I meant, we don't want the condition with hard coded conditions and output check, by using the more broader condition of allow_new_shape_infer limitation. (i.e., just allow_new_shape_infer should be enogh..)

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:482]: Got it! Updated. Thanks!

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 3]
```diff
@@ -476,6 +477,10 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, reorder_node
         if (fmt_prev.dimension() > 6 || fmt_next.dimension() > 6)
             return false;
 
+        // Skip reorder fusing to permute when allow_new_shape_infer is True
+        if (allow_new_shape_infer)
+            return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:482]: Sorry but please add one more condition. I'd like to add "input and output rank is different" to this. Could you please apply it? thanks!

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:482]: Updated! Thanks!

**Review by yeonbok — APPROVED** [review]

**Pull request merged**